src/agent/debuglet.ts

```diff
@@ -559,13 +570,14 @@ export class Debuglet extends EventEmitter {
       packageInfo.version;
     let desc = process.title + ' ' + mainScript;
 
+    // Incompatibilities!  Keys must be non-empty strings and can't contain ".", "#", "$", "/", "[", or "]"
     const labels: {[key: string]: string} = {
       'main script': mainScript,
-      'process.title': process.title,
+      'process title': process.title,
       'node version': process.versions.node,
       'V8 version': process.versions.v8,
-      'agent.name': packageInfo.name,
-      'agent.version': packageInfo.version,
+      'agent name': packageInfo.name,
```

Comment:
We should probably call this a breaking change, I'm assuming the issue is that firebase can't handle a `.` in keynames?

Perhaps we could use a `_` rather than a ` `, that reads a bit better to me and is easier to search.

Reply:
Correct; and I like the underscore idea.

This shouldn't be a breaking change (at least for users of the agent) but given the addition of the configuration and new dependencies this feature could be considered to be significant enough to warrant a major version bump.  Thoughts?

Reply:
Given all the restructuring, I might call it a breaking change, even if there's no one specific breakage we're calling attention to.